deepchecks/utils/html.py

```diff
@@ -10,49 +10,234 @@
+    if href:
+        if is_for_iframe_with_srcdoc and href.startswith('#'):
+            href = f'about:srcdoc{href}'
+        attrs = f'href={href} {attrs}'
+
+    if style:
+        attrs = f'{attrs} style="{style}"'
 
-    attrs = ' '.join([f'{k}="{v}"' for k, v in kwargs.items()])
     return f'<a {attrs}>{text}</a>'
+
+
+def details_tag(
+    title: str,
+    content: str,
+    id: t.Optional[str] = None,  # pylint: disable=redefined-builtin
+    attrs: t.Union[str, t.Mapping[str, str], None] = None,
+    style: t.Union[str, t.Mapping[str, str], None] = None,
+    content_attrs: t.Union[str, t.Mapping[str, str], None] = None,
+    content_style: t.Union[str, t.Mapping[str, str], None] = None,
+) -> str:
+    """Return HTML <details> tag."""
+    style = _stringify(style, 'style')
+    attrs = _stringify(attrs, 'attrs', template='{k}="{v}"')
+    content_style = _stringify(content_style, 'content_style')
+    content_attrs = _stringify(content_attrs, 'content_attrs', template='{k}="{v}"')
+
+    if id:
+        attrs = f'id="{id}" {attrs}'
+    if style:
+        attrs = f'{attrs} style="{style}"'
+    if content_style:
+        content_attrs = f'{content_attrs} style="{content_style}"'
+
+    return f"""
+        <details {attrs}>
+            <summary>{title}</summary>
+            <div {content_attrs}>
+            {content}
+            </div>
+        </details>
+    """
+
+
+def tabs_widget(data: t.Mapping[str, t.Union[str, t.List[str]]]) -> str:
+    tab_btn_template = '<button data-tab-index="{index}" onclick="deepchecksOpenTab(event)" {attrs}>{title}</button>'
+    tab_content_template = '<div class="deepchecks-tab" data-tab-index="{index}" {attrs}>{content}</div>'
+
+    buttons = []
+    tabs = []
+
+    for i, (k, v) in enumerate(data.items()):
+        if isinstance(v, list):
+            v = ''.join(v)
+        elif isinstance(v, str):
+            pass
+        else:
+            raise TypeError(f'Unsupported data value type - {type(v).__name__}')
+
+        attrs = 'open' if i == 0 else ''
+        buttons.append(tab_btn_template.format(index=i, title=k, attrs=attrs))
+        tabs.append(tab_content_template.format(index=i, content=v, attrs=attrs))
+
+    template = """
+    <div class="deepchecks-tabs">
+        <div class="deepchecks-tabs-btns">{buttons}</div>
+        {tabs}
+    </div>
+    <script>
+        function deepchecksOpenTab(event) {{
+            const btn = event.target;
+            if (btn.hasAttribute('open') === true)
+                return;
+            const tabsWidget = btn.closest('div.deepchecks-tabs');
+            const tabsBtns = btn.closest('div.deepchecks-tabs-btns');
+            const targetIndex = btn.getAttribute('data-tab-index');
+            tabsBtns.querySelectorAll('button[open]').forEach(it => it.removeAttribute('open'));
+            btn.setAttribute('open', '');
+            tabsWidget.querySelectorAll('div.deepchecks-tab[open]').forEach(it => it.removeAttribute('open'));
```

Comment:
doesn't we need some unique id in case there are multiple tabs widegts?

Reply:
hm/
no, we can extract btn which was clicked it tab-panel from the event

```javascript
   function deepchecksOpenTab(event) {
      const btn = event.target;
      if (btn.hasAttribute('open') === true)
         return;
      const tabsWidget = btn.closest('div.deepchecks-tabs');
     
```